component/common/loki/http/config.go

```diff
@@ -0,0 +1,86 @@
+// Package http contains a River serializable definition of the weaveworks server config in
+// https://github.com/weaveworks/common/blob/master/server/server.go#L62.
+package http
+
+import (
+	"flag"
+	"github.com/weaveworks/common/server"
+	"time"
+)
+
+// ServerConfig is a wrapper around server.ServerConfig.
+type ServerConfig struct {
```

Comment:
I think it's a good idea to expose fields defined in the Promtail server config, which just a subset applies here since there's things like `disabled`, `enable_runtime_reload` and other that don't apply to this case.
Exposing the whole of [weaveworks config](https://github.com/weaveworks/common/blob/master/server/server.go#L62) is too much in my opinion, and will be a PITA for users to configure through river.

I think a middle ground could be:
1. We expose the configs that are in the Promtail doc and apply to our case, which are:
```
http {
    [http_listen_address: <string>]
    [http_listen_port: <int> | default = 80]
    [http_server_read_timeout: <duration> | default = 30s]
    [http_server_write_timeout: <duration> | default = 30s]
    [http_server_idle_timeout: <duration> | default = 120s]
    [http_path_prefix: <string>] // could be ommited
}
grpc {
    [grpc_listen_address: <string>]
    [grpc_listen_port: <int> | default = 9095]
    [grpc_server_max_recv_msg_size: <int> | default = 4194304]
    [grpc_server_max_send_msg_size: <int> | default = 4194304]
    [grpc_server_max_concurrent_streams: <int> | default = 100]
}
[register_instrumentation: <boolean> | default = true] // can be ommited, since metrics are registered in the registered passed to target server
[graceful_shutdown_timeout: <duration> | default = 30s]
[log_level: <string> | default = "info"]
```
classified in their respective sub-block.

2. We do what you mentioned [here](https://github.com/grafana/agent/pull/3581#discussion_r1175542719) or requiring as argument for `NewTargetServer` the weaveworks config, so that component authors are able to tweak some specific setting available in weaveworks, but no in promtail. 
3. 
4. wdyt?

Reply:
Okay, let's start with these fields.

Some comments:
* As you said, let's remove the register_instrumentation and have it always be false. This will not be the 'main' flow HTTP server, and if we're passing the Registerer from the parent component, the metrics will be available on the root /metrics endpoint either way
* Let's use another default port for HTTP (9090 maybe?). Port 80 means that the server will need to run with superuser privileges which doesn't have to be the default